src/main/java/practice/StreamPractice.java

```diff
@@ -1,80 +1,66 @@
 package practice;
 
-import java.util.Collections;
+import java.util.Arrays;
+import java.util.Collection;
 import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
 import model.Candidate;
+import model.Cat;
 import model.People;
 
 public class StreamPractice {
-    /**
-     * Given list of strings where each element contains 1+ numbers:
-     * input = {"5,30,100", "0,22,7", ...}
-     * return min integer value. One more thing - we're interested in even numbers.
-     * If there is no needed data throw RuntimeException with message
-     * "Can't get min value from list: < Here is our input 'numbers' >"
-     */
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .map(n -> Arrays.stream(n.split(","))
+                        .map(Integer::parseInt)
+                        .filter(number -> number % 2 == 0)
+                        .min(Integer::compare))
+                .flatMap(Optional::stream)
+                .min(Integer::compare).orElseThrow(()
+                        -> new RuntimeException("Can't get min value from list: " + numbers));
     }
 
-    /**
-     * Given a List of Integer numbers,
-     * return the average of all odd numbers from the list or throw NoSuchElementException.
-     * But before that subtract 1 from each element on an odd position (having the odd index).
-     */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        double average = numbers.stream()
+                .mapToDouble(n -> numbers.indexOf(n) % 2 != 0 ? n - 1 : n)
```

Comment:
indexOf повертає перший індекс цього елементу в масиві - в цьому завданні краще починати з стріма індексів, а також тут не потрібно створювати додаткову змінну типу double - метод average робить все сам.
```suggestion
        IntStream.range(0, numbers.size())
                .mapToObj(i -> (i % 2 != 0) ? numbers.get(i) - 1 : numbers.get(i))
```